drivers/dht/dht.c

```diff
@@ -54,7 +56,9 @@ STATIC mp_obj_t dht_readinto(mp_obj_t pin_in, mp_obj_t buf_in) {
     mp_hal_pin_od_low(pin);
     mp_hal_delay_ms(18);
 
-    mp_uint_t irq_state = mp_hal_quiet_timing_enter();
+    if (mp_obj_is_true(irq_lock)) {
```

Comment:
mp_obj_is_true() is a little involved function so it would be worth calling it only once instead if tree times. You should consider creating a bool that will "cache" this computation once, something like:
bool irq_lock_true = mp_obj_is_true(irq_lock);

Reply:
well, I considered the call to mp_obj_is_true() as almost free. I can do that if you prefer.

Reply:
Check its implementation [here](https://github.com/micropython/micropython/blob/master/py/obj.c#L151). For most cases it is, indeed, but if someone passes some object there, it does require quite some checks.
Ultimately it's not my decision, I'm just a random guy doing a review, but I do think its worth saving those few CPU cycles.